wolfcrypt/src/asn.c

```diff
@@ -11250,6 +11250,67 @@ static int wc_SetCert_LoadDer(Cert* cert, const byte* der, word32 derSz)
     return ret;
 }
 
+#define ALTNAMESOFFSET (MAX_OCTET_STR_SZ + 2*MAX_SEQ_SZ+5)
+
+int wc_CertAddAltName_ex(Cert *cert, byte *name, int nameSz, cert_altname_type type) {
+	// Append encoded value from name into cert->altNames
+	byte *buf;
+	int i;
+	
+	if (cert->altNamesSz == 0) {
+		// Reserve space for header structure
+		cert->altNamesSz = ALTNAMESOFFSET;
+	}
+	if (cert->altNamesSz + nameSz + 3 > WC_CTC_MAX_ALT_SIZE)
+		return BUFFER_E;		// Input too large for buffer
+	if (type < ALTNAME_OtherName || type > ALTNAME_registeredID) 
+		return BAD_FUNC_ARG;
+	
+	buf = cert->altNames;  
+	i = cert->altNamesSz;		// Append new name at end of buffer
+
+	i += SetImplicit(0, type, nameSz, buf+i);
+	XMEMCPY(buf+i,name,nameSz);
+	cert->altNamesSz = i+nameSz;
+	return 0;
+}
+
+int wc_CertAddAltNameStr(Cert *cert, byte *name, cert_altname_type type) {
+	int nameSz;
+	nameSz = XSTRLEN(name);
+	if (type == ALTNAME_rfc822Name || type == ALTNAME_dNSName || type==ALTNAME_URI) {
+		return wc_CertAddAltName_ex(cert, name, nameSz, type);
+	} else {
+		return BAD_FUNC_ARG;
+	}
+}
+int wc_CertEncodeAltName(Cert *cert) {
+	// Encode cert->altNames field by pre-pending appropriate OID and header fields
+	byte hdrBuf[MAX_SEQ_SZ+MAX_OCTET_STR_SZ];
+	int i = 0;
+	int octSize, seqSize, namesSize;
+	byte subjectAltName[] = {0x06, 0x03, 0x55, 0x1D, 0x11};
```

Comment:
If this is a fixed sequence? If so consider adding to a header asn.h or asn_public.h would be fitting.